### PR TITLE
log: Use severity-based logging for leveldb/libevent messages, reverse LogPrintLevel order

### PR DESCRIPTION
**Pull request by laanwj**
Log messages from leveldb and libevent libraries in the severity+level based log format introduced in bitcoin/bitcoin#24464.

Example of messages before:
```
2022-05-24T18:11:57Z [libevent] libevent: event_add: event: 0x55da963fcc10 (fd 10), EV_READ    call 0x7f1c7a254620
2022-05-24T18:11:57Z [libevent] libevent: Epoll ADD(1) on fd 10 okay. Old events were 0; read change was 1 (add); write change was 0 (none); close change was 0 (none)
2022-05-24T18:12:08Z leveldb: Generated table #609127@1: 6445 keys, 312916 bytes
2022-05-24T18:12:08Z leveldb: Generated table #609128@1: 5607 keys, 268548 bytes
2022-05-24T18:12:08Z leveldb: Generated table #609129@1: 189 keys, 9384 bytes
2022-05-24T18:12:08Z leveldb: Generated table #609130@1: 293 keys, 13818 bytes
```

Example of messages after:
```
2022-05-24T17:59:52Z [libevent:debug] event_add: event: 0x5652f44dac10 (fd 10), EV_READ    call 0x7f210f2e6620
2022-05-24T17:59:52Z [libevent:debug] Epoll ADD(1) on fd 10 okay. Old events were 0; read change was 1 (add); write change was 0 (none); close change was 0 (none)
2022-05-24T17:59:53Z [leveldb:debug] Recovering log #1072
2022-05-24T17:59:53Z [leveldb:debug] Level-0 table #1082: started
2022-05-24T17:59:53Z [leveldb:debug] Level-0 table #1082: 193 bytes OK
2022-05-24T17:59:53Z [leveldb:debug] Delete type=3 #1070
2022-05-24T17:59:53Z [leveldb:debug] Delete type=0 #1072
```

The first commit changes it so that messages with level Warning and Error are always logged independent of the `-debug` setting. I think this is useful to make sure warnings and errors, which tend to be important, are not lost. In the future this should be made more configurable.

Last commit changes LogPrintLevel argument order to category, severity: This is more consistent with the other functions, as well as with the logging output itself. If we want to make this change, we should do it before it's all over the place.

### CODE REVIEW

File: src/logging.h [hunk 1]
```diff
@@ -210,7 +215,7 @@ static inline void LogPrintf_(const std::string& logging_function, const std::st
 
 #define LogPrintLevel(level, category, ...)               \
     do {                                                  \
-        if (LogAcceptCategory((category))) {              \
+        if (LogAcceptCategory((category, level))) {       \
```
Comment (jonatack) [inline src/logging.h:218]: At first glance it looks like the callers are using an inverted parameters order (level, category) compared to here.

Comment (laanwj) [reply src/logging.h:218]: I thought about that, and I'm fine with putting level first, but that means updating all callers of `LogAcceptCategory`. It's not that many, though.

Comment (laanwj) [reply src/logging.h]: It should build correctly now, the problem wasn't the order of arguments but invalid `()` nesting.

Comment (laanwj) [reply src/logging.h]: I agree it's strange that `LogPrintLevel` and `LogPrintLevel_`/`LogAcceptCategory` have inverted order of level, category arguments, but not sure it's in scope to fix here (nor in which direction to resolve it, I think that needs a separate discussion).

File: src/logging.h [hunk 2]
```diff
@@ -166,9 +166,14 @@ namespace BCLog {
 
 BCLog::Logger& LogInstance();
 
-/** Return true if log accepts specified category */
-static inline bool LogAcceptCategory(BCLog::LogFlags category)
+/** Return true if log accepts specified category, at the specified level. */
+static inline bool LogAcceptCategory(BCLog::LogFlags category, const BCLog::Level level = BCLog::Level::Debug)
```
Comment (MarcoFalke) [inline src/logging.h:170]: Wouldn't it be better to not have a default value and just inline the value once where it is needed?

**Review by maflcko — APPROVED** [review]

File: src/logging.h (hunk 2 above)
Comment (laanwj) [reply src/logging.h:170]: Sure, will do.

**Comment by laanwj** [comment]
Re-pushed [d2a8887→c4e7717](https://github.com/bitcoin/bitcoin/compare/d2a8887acea3df6d71f09d29fc6b73ef89d267c2..c4e77177276ea2b79c4675cb2678ee2cc757b743):

-  Remove default argument from `LogAcceptCategory`
- Add commit to change LogPrintLevel argument order to category, severity: This is more consistent with the other functions, as well as with the logging output itself. If we want to make this change, we should do it before it's all over the place.

**Review by jonatack — COMMENTED** [review]
Tested ACK c4e77177276ea2b79c4675cb2678ee2cc757b743

File: src/dbwrapper.cpp [hunk 3]
```diff
@@ -19,7 +19,7 @@ class CBitcoinLevelDBLogger : public leveldb::Logger {
     // This code is adapted from posix_logger.h, which is why it is using vsprintf.
     // Please do not do this in normal code
     void Logv(const char * format, va_list ap) override {
-            if (!LogAcceptCategory(BCLog::LEVELDB)) {
+            if (!LogAcceptCategory(BCLog::LEVELDB, BCLog::Level::Debug)) {
```
[the remark this answers is not in the document]
Comment (laanwj) [reply src/dbwrapper.cpp:22]: I agree, but as you say, this was the case before this PR, and is not introduced with category/severity based logging. There's probably tons of source files that, besides these ones, need `logging.h` included.

**Pull request merged**